sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -117,6 +118,152 @@ inline __SYCL_ALWAYS_INLINE
 
 } // namespace native
 
+namespace detail {
+
+template <typename T> struct is_bf16_storage_type {
+  static constexpr int value = false;
```

Comment:
Just curious, why you are not using `bool` or `auto` here?

Reply:
I don't know: this was added here: https://github.com/intel/llvm/pull/5748. I simply moved it to the correct experimental namespace in line with the extension document.  In any case it is deprecated already in favour of the `bfloat16` class implementation added in this PR and will be removed soon.